htdocs/product/class/api_products.class.php

```diff
@@ -1428,7 +1564,7 @@ public function getVariantsByProdRef($ref)
      *
      * @url POST {id}/variants
      */
-    public function addVariant($id, $weight_impact, $price_impact, $price_impact_is_percent, $features, $reference = false)
+    public function addVariant($id, $weight_impact, $price_impact, $price_impact_is_percent, $features, $reference = false, $ref_ext = '')
```

Comment:
There is a restler validator error with the argument $reference. It can be fixed by replacing:
line 1557 by: * @param  string $reference Customized reference of variant
line 1567 $reference = false : by $reference = ''